yt_dlp/extractor/youmaker.py

```diff
@@ -0,0 +1,430 @@
+        def fetch_page(page_number):
+            offset = page_number * page_size
+            info = self._call_api(
+                uid,
+                path='playlist/video',
+                what=f'playlist entries {offset + 1}-{offset + page_size}',
+                query={'playlist_uid': uid, 'offset': offset, 'limit': page_size})
+            if not isinstance(info, list):
+                raise ExtractorError('Unexpected playlist entries', uid, expected=False)
+
+            for item in info:
+                video_uid, title = itemgetter('video_uid', 'video_title')(item)
+                yield self.url_result(
+                    f'{self._base_url}/video/{video_uid}',
+                    ie=self.ie_key(),
+                    video_id=video_uid,
+                    video_title=title)
+
+        _ = self._categories  # preload categories
+        return OnDemandPagedList(fetch_page, page_size)
+
+    def _paged_channel_entries(self, uid, page_size=REQUEST_LIMIT):
+        def fetch_page(page_number):
+            offset = page_number * page_size
+            info = self._call_api(
+                uid,
+                path=f'video/channel/{uid}',
+                what=f'channel entries {offset + 1}-{offset + page_size}',
+                query={'offset': offset, 'limit': page_size})
+            if not isinstance(info, list):
+                raise ExtractorError('Unexpected channel entries', uid, expected=False)
+
+            for item in info:
+                video_uid, title = itemgetter('video_uid', 'title')(item)
+                self._cache[video_uid] = item
+                yield self.url_result(
+                    f'{self._base_url}/video/{video_uid}',
+                    ie=self.ie_key(),
+                    video_id=video_uid,
+                    video_title=title)
+
+        _ = self._categories  # preload categories
+        return OnDemandPagedList(fetch_page, page_size)
+
+    def _playlist_entries_by_id(self, uid):
+        _ = self._categories  # preload categories
+        info = self._call_api(uid, f'playlist/{uid}', what='playlist metadata')
+        return self.playlist_result(
+            self._paged_playlist_entries(info['playlist_uid']),
+            playlist_id=info['playlist_uid'],
+            playlist_title=info.get('name'),
+            playlist_description=None)
+
+    def _channel_entries_by_id(self, uid):
+        _ = self._categories  # preload categories
+        info = self._call_api(
+            uid,
+            path=f'video/channel/metadata/{uid}',
+            what='channel metadata')
+        return self.playlist_result(
+            self._paged_channel_entries(info['channel_uid']),
+            playlist_id=info['channel_uid'],
+            playlist_title=info.get('name'),
+            playlist_description=info.get('description'))
+
+    def _real_extract(self, url):
+        parsed_url = ParsedURL(url)
+        self._protocol = parsed_url.scheme
```

Comment:
Something I noticed across your various PR. Do not use class/instance variables for this. They should only be used when the property is meant to be shared b/w runs. If user passes 2 urls of same extractor, they are handled by the same instance of the extractor. So if it is being set in `_real_extract`, it most likely should just be a normal variable

Also, for this specific case, the protocol shouldn't depend on what user gave. Most extractors just use https, but if you want the user to have an option, obey `--prefer-insecure` using `self.http_scheme`